kv/task.go

```diff
@@ -296,6 +304,14 @@ func (s *Service) findTaskByOrg(ctx context.Context, tx Tx, filter influxdb.Task
 				return nil, 0, err
 			}
 
+			if filter.Type == nil {
+				ft := ""
+				filter.Type = &ft
+			}
+			if *filter.Type != influxdb.TaskTypeWildcard && *filter.Type != t.Type {
+				t = nil
+			}
+
 			// insert the new task into the list
 			if t != nil {
 				ts = append(ts, t)
```

Comment:
Take it or leave it suggestion. I found the flow a little hard to parse and a bit concerned about all the pointers. I would suggest something like:
```go
if t != nil {
  typ := ""
  if *filter.Type != nil {
    typ = *filter.Type
  }

  // if the filter type matches task type or filter type is a wildcard
  if typ == t.Type  || typ == influxdb.TakeTypeWildcard {
    ts = append(ts, t)
  }
}
```
Actually it appears in the original code that `t` could be nil. At least that nil check guard suggests it could be. So perhaps you do need a nil check on t somewhere first, otherwise the `t.Type` could nil pointer dereference. (either that or t is never really nil which is actually more than likely).

Reply:
good find. the call above `influxdb.IDFromString(string(v))` used to be able to return a nil task with no error, I am not sure it does that anymore. I do like the logic in your solution better, its more clean.